liblouis/internal.h

```diff
@@ -80,6 +80,34 @@ extern "C" {
 #define B15 0X4000
 #define B16 0X8000
 
+typedef struct intCharTupple {
+  int key;
+  char value;
+} intCharTupple;
+
+/**
+ * Mapping between braille dot and textual representation as used in dots operands
+ */
+const static intCharTupple dotMapping[] = {
+  {B1, '1'},
+  {B2, '2'},
+  {B3, '3'},
+  {B4, '4'},
+  {B5, '5'},
+  {B6, '6'},
+  {B7, '7'},
+  {B8, '8'},
+  {B9, '9'},
+  {B10, 'A'},
+  {B11, 'B'},
+  {B12, 'C'},
+  {B13, 'D'},
+  {B14, 'E'},
+  {B15, 'F'},
+  {B16, '0'},
```

Comment:
You made a mistake here. Every dot pattern has `B16` in it. It only "maps to" 0 if `B1` to `B15` are absent.

Reply:
Oh really, I thought there was something like that in the `utils.c` `showdots` code. let me check

Reply:
In `utils.c` in `_lou_showDots` it used to say:
``` C
if (dots[dotsPos] == B16)
    scratchBuf[bufPos++] = '0';
```
That's where I got it from.